cpp/data_structures/dynamic_tree/dynamic_tree.hpp

```diff
@@ -0,0 +1,414 @@
+/*
+    Dynamic Tree Node
+
+	-- NAME ---------- BEST --> WORST
+
+	length		    => o(1)
+
+	insert			=> o( (n log n) + (log n) + 1 )
+	search			=> o(1) --> o(log n)
+	sort            => o(n log n) --> o(n²)
+
+	getCopy        	=> o(1) --> o(log n)
+	replace   		=> o(1) --> o(log n)
+
+	remove			=> o(nodes * log n)
+    removeChild     => o(log n) --> o(nodes * log n)
+    removeChildren  =>
+
+    operator <      => o(1) 
+    operator >      => o(1)
+
+*/
+
+
+/*
+    Dynamic Tree 
+    
+    -- NAME ------	BEST --> WORST
+    search_up		o(1)
+    search_down		o(1) --> o(log n)
+
+    go_to			o(1) --> o(log n) 
+    go_to_root      o(1)
+    go_back         o(1)
+    travel_to	    o(log n) --> o(log n * path)
+
+*/
+
+#include <iostream>
+#include <algorithm>
+#include <vector>
+#include <string>
+
+#pragma once
+
+#ifndef DYNAMIC_TREE_H
+#define DYNAMIC_TREE_H
+
+namespace cpstl {
+
+    // ================================
+    // === Dynamic Tree Nodes class ===
+    // ================================
```

Comment:
The c++ standard for the multy line comment it is the following one
```suggestion
  /** 
    * Dynamic Tree Nodes class.
    **/
```